ui/components/component-library/form-text-field/form-text-field.tsx

```diff
@@ -0,0 +1,151 @@
+  HelpText,
+  HelpTextSeverity,
+  Label,
+  TextFieldSize,
+} from '..';
+import { PolymorphicRef } from '../box';
+import type { BoxProps } from '../box';
+import type { TextFieldProps } from '../text-field/text-field.types';
+import {
+  FormTextFieldSize,
+  FormTextFieldProps,
+  FormTextFieldComponent,
+} from './form-text-field.types';
+
+export const FormTextField: FormTextFieldComponent = React.forwardRef<
+  PolymorphicRef<any>,
+  FormTextFieldProps<any>
+>(
+  (
+    {
+      autoComplete,
+      autoFocus,
+      className = '',
+      defaultValue,
+      disabled,
+      isDisabled,
+      error,
+      helpText,
+      helpTextProps,
+      id,
+      inputProps,
+      inputRef,
+      label,
+      labelProps,
+      startAccessory,
+      maxLength,
+      name,
+      onBlur,
+      onChange,
+      onFocus,
+      placeholder,
+      readOnly,
+      required,
+      endAccessory,
+      size = FormTextFieldSize.Md,
+      textFieldProps,
+      truncate,
+      type = 'text',
+      value,
+      ...props
+    },
+    ref,
+  ) => {
+    // Validation function for 'id' prop
+    const validateIdProp = () => {
+      if (label && !id) {
+        throw new Error(
+          `If a label prop exists, you must provide an 'id' prop for the label's htmlFor attribute for accessibility. Warning coming from FormTextField component.`,
+        );
+      }
+    };
+
+    // Invoke the validation function
+    React.useEffect(() => {
+      validateIdProp();
+    }, [label, id]);
+
+    return (
+      <Box
+        className={classnames(
+          'mm-form-text-field',
+          {
+            'mm-form-text-field--disabled':
+              Boolean(isDisabled) || Boolean(disabled),
+          },
+          className,
+        )}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        ref={ref}
+        {...(props as BoxProps<any>)}
+      >
+        {label && (
+          <Label
+            htmlFor={id}
+            {...labelProps}
+            className={classnames(
+              'mm-form-text-field__label',
+              labelProps?.className ?? '',
+            )}
+          >
+            {label}
+          </Label>
+        )}
+        <TextField
+          className={classnames(
+            'mm-form-text-field__text-field',
+            textFieldProps?.className ?? '',
+          )}
+          size={size as unknown as TextFieldSize}
+          {...{
+            autoComplete,
+            autoFocus,
+            defaultValue,
+            disabled,
+            error,
+            id,
+            inputProps,
+            inputRef,
+            startAccessory,
+            maxLength,
+            name,
+            onBlur,
+            onChange,
+            onFocus,
+            placeholder,
+            readOnly,
+            required,
+            endAccessory,
+            truncate,
+            type,
+            value,
+            ...(textFieldProps as TextFieldProps<'input'>),
```

Comment:
I believe the wrapper is still a div
```suggestion
            ...(textFieldProps as TextFieldProps<'div'>),
```